luavm/src/lib/mod.rs

```diff
@@ -6,14 +6,21 @@ mod lua_value;
 use instructions::{arithmetic_operators::*, loads::*};
 use lua_value::{LuaNil, LuaValue};
 use luacompiler::bytecode::{instructions::opcode, LuaBytecode};
-use std::cell::RefCell;
+
+/// The instruction handler for each opcode.
+const OPCODE_HANDLER: &'static [fn(&Vm, u32) -> VmChange] =
+    &[mov, ldi, ldf, lds, add, sub, mul, div, modulus, fdiv, exp];
+
+/// Represents a pending update of the state of the vm.
+pub enum VmChange {
```

Comment:
I'm not sure I understand what this `enum` represents. Can you explain what it's doing to me in simple terms?

Reply:
I fear that we are trying to run before we walk. What we need is a very simple architecture where the registers are a `Vec`, and for instructions to modify them directly. That is the only way we'll ever get good performance, so we have to find a nice -- and, vitally, simple! -- design that lets us do that. Let's start with that fast, simple design and build complexity back up from there.

Reply:
Now I pass the vm mutably to the instructions, so they can modify the state directly. I am not sure why I didn't do this in the first place... Is this better?